src/jpegxl.imageio/jxlinput.cpp

```diff
@@ -57,15 +57,15 @@ class JxlInput final : public ImageInput {
     JxlResizableParallelRunnerPtr m_runner;
     std::unique_ptr<ImageSpec> m_config;  // Saved copy of configuration spec
     std::vector<uint8_t> m_icc_profile;
-    std::vector<float> m_pixels;
-    // std::vector<uint8_t> m_pixels;
+    uint8_t* m_buffer;
```

Comment:
I think it would be better to use a `std::unique_ptr<uint8_t[]> m_buffer` and know that we can't possibly leak.

Reply:
I did unique_ptr. Please check, i'm not so well familiar with them.